sdk/elixir/dagger_codegen/lib/dagger/codegen/elixir_generator/formatter.ex

```diff
@@ -0,0 +1,152 @@
+defmodule Dagger.Codegen.ElixirGenerator.Formatter do
+  alias Dagger.Codegen.Introspection.Types.TypeRef
+
+  def format_module("Query"), do: format_module("Client")
+
+  def format_module(name) do
+    Module.concat(Dagger, Macro.camelize(name))
+    |> to_string()
+    |> String.trim_leading("Elixir.")
+  end
+
+  def format_var_name("Query"), do: format_var_name("Client")
+
+  def format_var_name(name) do
+    Macro.underscore(name)
+  end
+
+  # Temporarily fixes for issue https://github.com/dagger/dagger/issues/6310.
+  @acronym_words %{
+    "GPU" => "Gpu",
+    "VCS" => "Vcs"
+  }
+
+  def format_function_name(name, acronym_words \\ @acronym_words) do
```

Comment:
Must make sure to handle possible reserved words in the language, I couldn't spot this being done.

Maybe an "end" function argument, or a "do" function name. Anywhere where there could be a conflict, most importantly that would lead to a syntax error, but can also do it for soft keywords as well (shadowing built-ins). Example in Python:

https://github.com/dagger/dagger/blob/795a73595d8dd0c14bc9b71e6a137852e7e0e9ac/sdk/python/src/dagger/_codegen/generator.py#L345-L346

With modules, codegen is open to many end user choices, it's not just core. But even in the core API, I've already seen us choosing a different name than the one desidered due to a conflict with the codegen in an experimental SDK. It was easier to choose a different name then, but it's better to make this a requirement for SDKs going forward.

Reply:
I see. Adding `_` suffix to align with Python is interesting. Add the issue https://github.com/dagger/dagger/issues/6940 to follow up later. 🙇